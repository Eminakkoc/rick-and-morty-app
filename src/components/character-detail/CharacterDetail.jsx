import React from 'react';
import PropTypes from 'prop-types';

import { LanguageConsumer } from 'common/language/LanguageContext';

import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';

import InfoPair from 'components/info-pair/InfoPair';

import languages from 'common/language/languages';

const CharacterDetail = (props) => {
  const {
    image,
    name,
    origin,
    episodeList,
  } = props;

  const getEpisodes = episodes => (
    <div className="episode-list-container">
      {
        episodes && episodes.map(episode => (
          <div className="episode">
            { episode }
          </div>
        ))
      }
    </div>
  );

  return (
    <LanguageConsumer>
      {({ language }) => (
        <Card className="details-card">
          <div className="details-card-wrapper">
            <CardContent>
              <div className="details-container">
                <div className="details-image-container">
                  <img className="details-image" src={image} alt="" />
                </div>
                <div className="details-wrapper">
                  <div className="character-name">
                    {name}
                  </div>
                  <div className="details-character-info">
                    <InfoPair
                      label={languages[language].character_details_origin_label}
                      value={origin}
                    />
                    <InfoPair
                      label={languages[language].character_details_episodes_label}
                      value={getEpisodes(episodeList)}
                    />
                  </div>
                </div>
              </div>
            </CardContent>
          </div>
        </Card>
      )}
    </LanguageConsumer>
  );
};

CharacterDetail.propTypes = {
  image: PropTypes.string,
  name: PropTypes.string,
  origin: PropTypes.string,
  episodeList: PropTypes.arrayOf(PropTypes.string),
};

CharacterDetail.defaultProps = {
  name: '',
  image: null,
  origin: null,
  episodeList: null,
};

export default CharacterDetail;
