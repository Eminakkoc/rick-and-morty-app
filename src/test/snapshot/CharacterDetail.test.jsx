// CharacterDetail.test.js
import React from 'react';
import CharacterDetail from '../../components/character-detail/CharacterDetail';
import renderer from 'react-test-renderer';

test('Character Detail Component Structure', () => {
  const component = renderer.create(
    <CharacterDetail />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
