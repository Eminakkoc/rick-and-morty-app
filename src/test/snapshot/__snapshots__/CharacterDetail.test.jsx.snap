// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Detail Component Structure 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root details-card MuiPaper-rounded"
>
  <div
    className="details-card-wrapper"
  >
    <div
      className="MuiCardContent-root"
    >
      <div
        className="details-container"
      >
        <div
          className="details-image-container"
        >
          <img
            alt=""
            className="details-image"
            src={null}
          />
        </div>
        <div
          className="details-wrapper"
        >
          <div
            className="character-name"
          >
            
          </div>
          <div
            className="details-character-info"
          >
            <div
              className="info-pair-container character-origin"
            >
              <div
                className="info-pair-key"
              >
                Origin: 
              </div>
              <div
                className="info-pair-value"
              />
            </div>
            <div
              className="info-pair-container character-epsiodes"
            >
              <div
                className="info-pair-key"
              >
                Episodes: 
              </div>
              <div
                className="info-pair-value"
              >
                <div
                  className="episode-list-container"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
